### PR TITLE
[4.0] Reworking result layout of com_finder

### PR DESCRIPTION
**Pull request by Hackwar**
The result layout of com_finder is very basic and the results of com_search leave a bigger impression. If we want all that information to be presented in the search results, is another question.

This PR reworks the result layout of com_finder to look more like those of com_search. I'm not happy with the current result myself, but I wanted to bring this up for discussion. What do you guys think? Should we rework this or rather not? Or rather just remove the URL below the result? (See #20571)

### CODE REVIEW

**Comment by brianteeman** [comment]
I like the idea that both search components should have the same layout. This pr is close but still not identical

### Search
chrome_2018-05-25_12-33-41

### Smart search
chrome_2018-05-25_12-34-38

File: components/com_finder/tmpl/search/default_result.php [hunk 1]
```diff
@@ -46,22 +46,44 @@
...
-		<div class="small result-url">
-			<?php echo $this->baseUrl, JRoute::_($this->result->route); ?>
-		</div>
-	<?php endif; ?>
-</li>
+	</dd>
+<?php endif; ?>
+<?php if ($this->result->created && $this->params->get('show_date', 1)) : ?>
+	<dd class="result-created">
+		<?php echo JText::sprintf('JGLOBAL_CREATED_DATE_ON', \JHtml::_('date', $this->result->created, \JText::_('DATE_FORMAT_LC3'))); ?>
```
Comment (SharkyKZ) [inline components/com_finder/tmpl/search/default_result.php]: Add backslash before `JText::sprintf`.

File: plugins/finder/content/content.php [hunk 2]
```diff
@@ -347,7 +347,7 @@ protected function getListQuery($query = null)
 		// Check if we can use the supplied SQL query.
 		$query = $query instanceof DatabaseQuery ? $query : $db->getQuery(true)
 			->select('a.id, a.title, a.alias, a.introtext AS summary, a.fulltext AS body')
-			->select('a.state, a.catid, a.created AS start_date, a.created_by')
+			->select('a.state, a.catid, a.created AS start_date, a.created, a.created_by')
```
Comment (wilsonge) [inline plugins/finder/content/content.php]: Do we really need to add `a.created` when we're already selecting it just calling it `start_date`?

Comment (Hackwar) [reply plugins/finder/content/content.php]: You are right. Reverting that one.

File: components/com_finder/tmpl/search/default_result.php (hunk 1 above)
Comment (carlitorweb) [reply components/com_finder/tmpl/search/default_result.php]: Why not use from Text class? Text::sprintf

**Comment by carlitorweb** [comment]
Well, I really not like the layout of com_search. I Like more the compact mode of com_finder, Title, Text and maybe the URL.
So, I test this PR and work ok, but not sure if this PR is more like an RFC or for add this reworking layout.

**Comment by Hackwar** [comment]
I will see what I can do to incorporate your comment @carlitorweb

**Comment by carlitorweb** [comment]
Thank you :). Is just my opinion about it.

**Comment by carlitorweb** [comment]
Some update for this one @Hackwar ?

**Comment by Hackwar** [comment]
I will. I just need some more time. The priority of this PR is rather low.

**Comment by carlitorweb** [comment]
> The priority of this PR is rather low.

Yes I know

**Comment by Hackwar** [comment]
I've updated this PR and consider this now feature-complete. I would call this an passable generalised result layout, which now can display the different taxonomies, the date of the item and the URL besides the title and the description. If somebody has a good solution to get a spacer between the search results, I'm all ears.

While implementing the taxonomies, it became apparent that we want to control which of these to display and which to hide. In theory the taxonomies support publish/unpublish states and even the access modifiers. However right now the taxonomies are stored inside the serialized FinderIndexerResult object and are not updated when retrieving the search results or when changing the taxonomies in the backend. Besides that, the taxonomies are lacking a GUI to modify the access modifiers. I plan to rework the taxonomies, too, and would address those issues there.

Happy testing.

**Review by brianteeman — APPROVED** [review]
approve langauge strings

**Comment by carlitorweb** [comment]
**BEFORE**

screenshot_20180706091424

**AFTER**

screenshot_20180706091412

The highlighted word inside a `badge` no looks so good. My personal opinion, I hate that "highlight" stuff in a search result. For me, a clean result page as much as possible can be, is better for the user can find faster what they looking for. 

A way I like to see a search results page is having, for example, something like this:

screenshot_20180706093417

Can be rows with 2 column or 3 depending on the layout.

**Comment by Hackwar** [comment]
@carlitorweb I agree sort of with you. This is definitely not the ultimate search result layout, but I guess it is the best compromise that we can have here. Anyway, thanks for testing. I will fix the conflict and then it would be nice if you could mark this as successfully tested. :wink:

**Comment by carlitorweb** [comment]
> This is definitely not the ultimate search result layout

I know buddy, just was sharing with you my thought. Waiting for the commit, for put the test.

**Comment by Hackwar** [comment]
Conflicts fixed.

**Comment by carlitorweb** [comment]
I have tested this item :white_check_mark: successfully on 4dea3663d373ac7b43919b3050da183e97dfe1fc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20572">issues.joomla.org/tracker/joomla-cms/20572</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 4dea3663d373ac7b43919b3050da183e97dfe1fc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20572">issues.joomla.org/tracker/joomla-cms/20572</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20572">issues.joomla.org/tracker/joomla-cms/20572</a>.</sub>

**Comment by Hackwar** [comment]
Thank you!

**Comment by Hackwar** [comment]
Fixed the conflict by the merge of #20681. Do we have to retest all of this or can we simply merge this?

**Comment by TobsBobs** [comment]
screenshot_2018-07-07 home

**Comment by TobsBobs** [comment]
I have tested this item :white_check_mark: successfully on b1787b64a7cac582ba777e57ef8ddddf0e64bab8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20572">issues.joomla.org/tracker/joomla-cms/20572</a>.</sub>

**Comment by wilsonge** [comment]
Thanks :)

**Pull request merged**